stream/src/main/scala/org/apache/pekko/stream/impl/ActorMaterializerImpl.scala

```diff
@@ -62,7 +62,8 @@ import pekko.util.OptionVal
     val effectiveProps = props.dispatcher match {
       case Dispatchers.DefaultDispatcherId =>
         // the caller said to use the default dispatcher, but that can been trumped by the dispatcher attribute
-        props.withDispatcher(context.effectiveAttributes.mandatoryAttribute[ActorAttributes.Dispatcher].dispatcher)
+        val dispatcher = context.effectiveAttributes.mandatoryAttribute[ActorAttributes.Dispatcher].dispatcher
+        props.withDispatcher(dispatcher)
```

Comment:
can this be reverted?

Reply:
Why?

Reply:
Seems I missed the mailbox part

Reply:
the new code is the same as the old code - just 2 lines and a new 'val' - can't it just be left as it was?

Reply:
line 65 and line 66 can be merged，no need to define the `val dispathcer`

Reply:
Inline is too long, But if you both prefer, I can revert to old style.

I can do that tomorrow.